xarray/backends/pydap_.py

```diff
@@ -46,6 +46,7 @@ def _getitem(self, key):
         # downloading coordinate data twice
         array = getattr(self.array, "array", self.array)
         result = robust_getitem(array, key, catch=ValueError)
+        result = np.asarray(result)
```

Comment:
Otherwise we were returning `pydap.BaseType` and relied on an `np.asarray` call in `explicit_indexing_adapter`. I've removed that call below.

I think this is a improvement since now all backend types *should* be returning duck arrays on `_getitem`